app/modules/finance/operations_router.py

```
@@ -48,13 +52,35 @@ def create_payment_voucher(payload:dict,db:Session=Depends(get_db),principal:Pri
-    db.add(m.FinanceApproval(school_id=principal.school_id,entity_type='payment_voucher',entity_id=r.id,sequence=sequence,required_role='admin',approver=principal.user_id,decision=decision,amount_threshold=r.amount,reason='Voucher approval',decided_at=datetime.now(timezone.utc)));r.status='APPROVED' if decision=='APPROVE' else 'REJECTED';_audit(db,principal,'approve','payment_voucher',r.id,f'Voucher {r.voucher_number}: {decision}');db.commit();db.refresh(r);return r
+    if r.status in {'REJECTED','PAID'}:raise HTTPException(409,f'Voucher is already {r.status.lower()}')
+    required=_required_approvals(_money(r.amount)); existing=db.query(m.FinanceApproval).filter_by(school_id=principal.school_id,entity_type='payment_voucher',entity_id=r.id).all()
+    if any(a.sequence==sequence for a in existing): raise HTTPException(409,f'Approval sequence {sequence} has already been decided')
+    approved_count=sum(1 for a in existing if a.decision=='APPROVE')
+    if sequence != approved_count+1: raise HTTPException(409,f'Next required approval sequence is {approved_count+1}')
+    if sequence>required: raise HTTPException(400,f'This voucher requires only {required} approval(s)')
+    db.add(m.FinanceApproval(school_id=principal.school_id,entity_type='payment_voucher',entity_id=r.id,sequence=sequence,required_role='admin',approver=principal.user_id,decision=decision,amount_threshold=r.amount,reason='Voucher approval',decided_at=datetime.now(timezone.utc)))
+    if decision=='REJECT': r.status='REJECTED'
+    elif approved_count+1>=required: r.status='APPROVED'
+    else: r.status='PENDING_APPROVAL'
+    _audit(db,principal,'approve','payment_voucher',r.id,f'Voucher {r.voucher_number}: {decision} sequence {sequence}/{required}');db.commit();db.refresh(r);return r
```

Ruff (0.16.1)

[error] Multiple statements on one line (semicolon)

(E702)